hub/core/chunk_engine/write.py

```diff
@@ -105,12 +35,15 @@ def write_bytes(
             end_byte: End byte for this sample. Will be equal to the length of the last chunk written to.
     """
 
+    # TODO: `_get_last_chunk(...)` is called during an inner loop. memoization here OR having an argument is preferred for performance
```

Comment:
So... we want a cache? :facepalm:

Reply:
i mean yeah but this is different because we need to memoize the `bytearray`ed chunk

Reply:
or pass it in as a parameter from outside of the loop